logstash-core/src/test/java/org/logstash/instrument/metrics/counter/LongCounterTest.java

```diff
@@ -0,0 +1,60 @@
+package org.logstash.instrument.metrics.counter;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.logstash.instrument.metrics.MetricType;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+/**
+ * Unit tests for {@link LongCounter}
+ */
+public class LongCounterTest {
+
+    private final long INITIAL_VALUE = 99l;
+    private LongCounter longCounter;
+
+    @Before
+    public void _setup() {
+        longCounter = new LongCounter(Collections.singletonList("foo"), "bar", INITIAL_VALUE);
+    }
+
+    @Test
+    public void getValue() {
+        assertThat(longCounter.getValue()).isEqualTo(INITIAL_VALUE);
+    }
+
+    @Test
+    public void increment() {
+
+        longCounter.increment();
+        assertThat(longCounter.getValue()).isEqualTo(INITIAL_VALUE + 1);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void incrementByNegativeValue() {
+        longCounter.increment(-100l);
+    }
+
+    @Test
+    public void incrementByValue() {
+        longCounter.increment(100l);
+        assertThat(longCounter.getValue()).isEqualTo(INITIAL_VALUE + 100);
+    }
+
+    @Test
+    public void noInitialValue() {
+        LongCounter counter = new LongCounter(Collections.singletonList("foo"), "bar");
+        counter.increment();
+        assertThat(counter.getValue()).isEqualTo(1l);
+    }
+
+    @Test
+    @SuppressWarnings( "deprecation" )
```

Comment:
This is a deprecation warning. To keep the API (mostly) passive to Ruby, I default implemented a couple of Ruby methods (type, get, inspect) in the Metrics interface and deprecated them so that Java consumers won't use them.